testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java

```diff
@@ -0,0 +1,69 @@
+package com.baeldung.optional;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class OptionalUnitTest {
+
+    private final String stringToTest = "testString";
+
+    @Test
+    public void givenOptionalWithValue_whenAssertThatIsNotEmpty_thenOk() {
+        assertThat(optionalWithValueMethodToTest()).isNotEmpty();
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertThatHasValue_thenOk() {
+        assertThat(optionalWithValueMethodToTest()).hasValue(stringToTest);
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertEqualsOptionalObject_thenOk() {
+        assertEquals(Optional.of(stringToTest), optionalWithValueMethodToTest());
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertEqualsGet_thenOk() {
+        Optional<String> optional = optionalWithValueMethodToTest();
+        assertEquals(stringToTest, optional.get());
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertTrueIsPresentAndGet_thenOk() {
+        Optional<String> optional = optionalWithValueMethodToTest();
+        assertTrue(optional.isPresent() && stringToTest.equals(optional.get()));
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenIsPresentAndGetSplit_thenOk() {
+        Optional<String> optional = optionalWithValueMethodToTest();
+
+        assertTrue(optional.isPresent());
+        assertEquals(stringToTest, optional.get());
+    }
+
+    @Test
+    public void givenEmptyOptional_whenAssertThatIsEmpty_thenOk() {
+        assertThat(optionalIsEmptyMethodToTest()).isEmpty();
+    }
+
+    @Test
+    public void givenEmptyOptional_whenAssertIsNotPresent_thenOk() {
+        Optional<String> optional = optionalIsEmptyMethodToTest();
+        assertFalse(optional.isPresent());
+    }
+
+    private Optional<String> optionalWithValueMethodToTest() {
+        return Optional.of(stringToTest);
+    }
```

Comment:
You've `DRY`ed out the tests by doing extract method here of a common bit of data, but it makes the test harder to follow. Inline these. In general, a trivial bit of code like this is easier re-written every time in a test, and that's especially true for example snippets in articles.

Reply:
I agree but I was on to follow the StackOverflow example

Reply:
Let's make our own version of the example that suits our needs.

SO examples aren't always the best ideas ;)